dissect/target/plugins/os/unix/log/audit.py

```diff
@@ -1,60 +1,82 @@
-    def check_compatible(self):
-        path = self.target.fs.path("/var/log/audit/")
+    def __init__(self, target):
+        super().__init__(target)
+        self.log_paths = self.get_log_paths()
+
+    @internal
+    def get_log_paths(self) -> list[Path]:
+        default_config = "/etc/audit/auditd.conf"
+        paths = ["/var/log/audit/audit.log"]
+
+        if (path := self.target.fs.path(default_config)).exists():
+            for line in path.open("rt"):
+                line = line.strip()
+                if not line:
+                    continue
+                if "log_file" in line:
+                    log_file = line.split("=")[-1].strip()
+                    paths.append(log_file)
 
-        if path.exists():
-            return True
-        else:
-            return False
+        return paths
+
+    def check_compatible(self) -> bool:
+        return self.target.fs.path("/var/log/audit/").exists()
 
     @export(record=[AuditRecord])
-    def audit(self):
-        """Return information stored in /var/log/audit.
+    def audit(self) -> Iterator[AuditRecord]:
+        """Return CentOS and RedHat audit information stored in /var/log/audit*.
 
         The audit log file on a Linux machine stores security-relevant information.
-        Based on pre-configured rules.
+        Based on pre-configured rules. Log messages consist of space delimited key=value pairs.
 
         Sources:
             - https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/security_guide/chap-system_auditing
             - https://linux-audit.com/linux-audit-log-files-in-var-log-audit/
+            - https://man7.org/linux/man-pages/man8/auditd.8.html
+            - https://man7.org/linux/man-pages/man8/ausearch.8.html
+            - https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-understanding_audit_log_files
         """  # noqa: E501
-        audit_type = re.compile(r"(?<=type=).+?\W")
-        audit_msg = re.compile(r"(?<=\):\W).+")
-        audit_meta = re.compile(r"(?<=msg=audit\().+?\)")
-
-        for file in self.target.fs.listdir_ext("/var/log/audit"):
-            fh = file.open("rt")
-
-            for line in fh:
-                # group returns the string representation of the match when using re.search
-                _type = audit_type.search(line).group().strip(" ")
-                msg = audit_msg.search(line).group()
-                audit = audit_meta.search(line).group()
-                ts, _id = audit.split(":")
-
-                yield AuditRecord(
-                    ts=datetime.datetime.fromtimestamp(float(ts)),
-                    type=_type,
-                    id=int(_id.strip(")")),
-                    msg=msg,
-                    auditlog=None,
-                    _target=self.target,
-                )
+
+        for path in self.log_paths:
+            print(path)
+            for file in self.target.fs.path(path).parent.glob("*"):
```

Comment:
Is this glob not a bit too generic? What if it's configured to log to `/var/log/audit.log`, then this would read all log files in `/var/log`.

Reply:
Improved in [f10e87a](https://github.com/fox-it/dissect.target/pull/173/commits/f10e87adc350cba6a0c720fe32bdc9b1bd16bfd4):

```
self.target.fs.path(path).parent.glob(basename(path) + "*")
```